imagebuildah/build.go

```diff
@@ -318,6 +318,37 @@ func buildDockerfilesOnce(ctx context.Context, store storage.Store, logger *logr
 		}
 		mainNode.Children = append(mainNode.Children, additionalNode.Children...)
 	}
+
+	// Check if any modifications done to labels
+	// add them to node-layer so it becomes regular
+	// layer.
+	// Reason: Docker adds label modification as
+	// last step which can be processed as regular
+	// steps and if no modification is done to layers
+	// its easier to re-use cached layers.
+	if len(options.Labels) > 0 {
+		for _, labelSpec := range options.Labels {
+			label := strings.SplitN(labelSpec, "=", 2)
+			labelLine := ""
+			key := label[0]
+			value := ""
+			if len(label) > 1 {
+				value = label[1]
+			} else {
+				value = ""
+			}
+			labelLine = fmt.Sprintf("LABEL %s=%s\n", key, value)
+			if len(labelLine) > 1 {
```

Comment:
When would this be `false`?

Reply:
yeah, might be better to check if key and value don't equal ""

Reply:
I think this is needed for basic validation since `--label` accepts arbitrary string similar check is also present in executor itself.

Reply:
Right, but isn't labelLine at a minimum going to contain "LABEL = "?  That would never be 0 or less right?

Reply:
@TomSweeneyRedHat oh right thanks changing this to `key != ""`, we can support empty value since docker supports empty values (manually verified).